openlibrary/plugins/upstream/models.py

```diff
@@ -825,14 +823,19 @@ def can_undo(self):
         return self.get_undo_changeset() is None
 
     def process_docs_before_undo(self, docs):
-        works = [doc for doc in docs if doc['key'].startswith("/works/")]
-        for w in works:
-            if w.get("authors"):
+        # FIXME: Is this incorrect in the case of undoing a merge?
+        # the code below checks if authors are currently redirects in ctx.site
+        # but if the redirect is incorrect and being reverted in this changeset,
+        # the code below will update all the books to have the incorrect target
+        # author...
```

Comment:
@mekarpeles can you check this logic here, I think this whole method is unnecessary and is preventing undos rfom working, but am worried I'm missing something.  If this hook is to do anything, it should revert all author changes first, then allow the books to be reverted, but it is not set up to work like that.

I think the site will prevent work saves that reference a redirect author, so an Author merge undo will need to first revert the author redirects and save, then undo the works/editions to the previous correct (now not a redirect) author.